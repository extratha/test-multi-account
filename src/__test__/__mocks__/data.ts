export const data = {
  test: 'test'
}