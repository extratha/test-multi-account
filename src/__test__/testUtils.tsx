import { RenderOptions, RenderResult, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { SWRConfig } from 'swr';
import { theme } from '@/config/config-mui/theme';
import { ThemeProvider as MUIThemeProvider } from '@mui/material/styles';
import { NextIntlClientProvider } from 'next-intl';
import commonTh from '../../public/locales/th/common.json';
import commonEn from '../../public/locales/th/common.json';
type RenderWithProvider = (
  elm: React.ReactElement,
  renderOptions?: ProviderOptions,
) => RenderResult;

export type ProviderOptions = {
  formOptions?: Record<string, unknown>;
} & RenderOptions;

const messages = {
  ...commonTh,
  ...commonEn,
};

const renderWithProviders: RenderWithProvider = (
  component,
  renderOptions?,
  locale = 'th',
) => {
  const wrapper = ({ children }: { children: React.ReactNode }) => (
    <NextIntlClientProvider locale={locale} messages={messages}>
      <MUIThemeProvider theme={theme}>
        <SWRConfig value={{ provider: () => new Map() }}>{children}</SWRConfig>
      </MUIThemeProvider>
    </NextIntlClientProvider>
  );

  return render(component, {
    wrapper,
    ...renderOptions,
  });
};

export { renderWithProviders as render, userEvent };
export * from '@testing-library/react';
