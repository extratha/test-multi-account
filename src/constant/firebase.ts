export const remoteConfigKey = {
  SYMPTOM_CHECKER: "SymptomChecker",
  LAB_INTERPRET_REQUIRE_FIELDS: "LabInterpretRequireFields",
};
