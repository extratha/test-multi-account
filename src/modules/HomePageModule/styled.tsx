import { Stack, Typography, styled } from "@mui/material";

export const ContentContainerWrapper = styled(Stack)(({ theme }) => ({
  backgroundColor: theme.palette.background.paper,
  padding: "1rem",
  borderRadius: "20px",
  height: "100%",
  overflowY: "auto",
}));

export const ContentContainer = styled(Stack)({
  height: "100%",
  maxHeight: "100vh",
  width: "100%",
  padding: "1rem",
});

export const TypographyPageHeadline = styled(Typography)(() => ({
  width: "fit-content",
  background: "linear-gradient(90deg, #023781 0%, #028DF5 50%, #01EAFF 100%)",
  WebkitBackgroundClip: "text",
  WebkitTextFillColor: "transparent",
  backgroundClip: "text",
  textFillColor: "transparent",
}));
