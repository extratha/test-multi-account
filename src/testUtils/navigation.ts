import * as NextNavigation from "next/navigation";

jest.mock("next/navigation", () => ({
  useRouter: jest.fn(),
  useParams: jest.fn(),
  usePathname: jest.fn(),
  useSearchParams: jest.fn(),
  useServerInsertedHTML: jest.fn(),
}));

export interface SpyUseSearchParams {
  get: jest.Mock;
}

export interface SpyUseRouter {
  replace: jest.Mock;
  push: jest.Mock;
  back: jest.Mock;
}

export const spyUseParams = (params: any = {}) => {
  jest.spyOn(NextNavigation, "useParams").mockReturnValue(params);
  return NextNavigation.useParams as jest.Mock;
};

export const spyUseRouter = (): SpyUseRouter => {
  const replace = jest.fn();
  const push = jest.fn();
  const back = jest.fn();
  jest.spyOn(NextNavigation, "useRouter").mockReturnValue({ replace, push, back } as any);
  return { replace, push, back };
};

export const spyUsePathname = (pathname: string) => {
  jest.spyOn(NextNavigation, "usePathname").mockReturnValue(pathname);
};

export const spyUseSearchParams = (): SpyUseSearchParams => {
  const get = jest.fn();
  jest.spyOn(NextNavigation, "useSearchParams").mockReturnValue({ get } as any);
  return {
    get,
  };
};

export const spyUseServerInsertedHTML = () => {
  return jest.spyOn(NextNavigation, "useServerInsertedHTML");
};
