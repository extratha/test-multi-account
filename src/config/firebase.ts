// Import the functions you need from the SDKs you need
import { envConfig } from "@/constant/env";
import { initializeApp } from "firebase/app";

// TODO: Add SDKs for Firebase products that you want to use
// https://firebase.google.com/docs/web/setup#available-libraries

// Your web app's Firebase configuration
// For Firebase JS SDK v7.20.0 and later, measurementId is optional
const firebaseConfig = {
  apiKey: envConfig.firebase.apiKey,
  authDomain: envConfig.firebase.authDomain,
  projectId: envConfig.firebase.projectId,
  storageBucket: envConfig.firebase.storageBucket,
  messagingSenderId: envConfig.firebase.messagingSenderId,
  appId: envConfig.firebase.appId,
  measurementId: envConfig.firebase.measurementId,
};

const firebaseApp = initializeApp(firebaseConfig);
export default firebaseApp;
