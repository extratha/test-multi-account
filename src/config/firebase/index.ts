// Import the functions you need from the SDKs you need
import { initializeApp, getApps } from "firebase/app";
import { getAnalytics } from "firebase/analytics";
import { staticEnvConfig } from "@/constant/env";

// TODO: Add SDKs for Firebase products that you want to use
// https://firebase.google.com/docs/web/setup#available-libraries

// Your web app's Firebase configuration
// For Firebase JS SDK v7.20.0 and later, measurementId is optional
const firebaseConfig = {
  apiKey: staticEnvConfig.firebase.apiKey,
  authDomain: staticEnvConfig.firebase.authDomain,
  projectId: staticEnvConfig.firebase.projectId,
  storageBucket: staticEnvConfig.firebase.storageBucket,
  messagingSenderId: staticEnvConfig.firebase.messagingSenderId,
  appId: staticEnvConfig.firebase.appId,
  measurementId:staticEnvConfig.firebase.measurementId
};

// Initialize Firebase
export const app = getApps().length === 0 ? initializeApp(firebaseConfig) : getApps()[0];

export const analytics = (typeof window !== "undefined") ? getAnalytics(app) : undefined;