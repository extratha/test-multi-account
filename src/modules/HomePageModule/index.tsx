"use client";
import { Grid, Card as MuiCard, Stack, styled, Typography } from "@mui/material";
import { useRouter } from "next/navigation";
import { useEffect, useState } from "react";

import { IconAiInterpret } from "@/assets";
import DashboardPage from "@/components/Page/DashboardPage";
import PageTitle from "@/components/Typography/PageTitle";
import useTranslation from "@/locales/useLocale";
import { usePageLoadingStore } from "@/store";
import { DashboardMenuConfigResult, HomeMenuItemConfig } from "@/types/model.ui";
import { getDashboardMenuConfig } from "@/utils/firebase";

const Card = styled(MuiCard)(({ theme }) => ({
  cursor: "pointer",
  display: "flex",
  flexDirection: "column",
  alignItems: "center",
  textAlign: "center",
  padding: "16px",
  boxShadow: "none",
  borderRadius: "12px",
  border: `1px solid ${theme.palette.background.border}`,
}));

const HomePageModule = () => {
  const { translation } = useTranslation();
  const { setPageLoading } = usePageLoadingStore();
  const router = useRouter();
  const [config, setConfig] = useState<DashboardMenuConfigResult>();

  const handleClickMenuItem = (item: HomeMenuItemConfig) => {
    router.replace(item.path);
  };

  const fetchConfig = async () => {
    try {
      setPageLoading(true);
      const configData = await getDashboardMenuConfig();

      setConfig(configData);
      setPageLoading(false);
    } catch (error) {
      setPageLoading(false);
    }
  };

  useEffect(() => {
    fetchConfig();
  }, []);

  return (
    <>
      {config && (
        <DashboardPage>
          <Stack spacing="16px" alignItems="flex-start" padding="40px">
            <PageTitle>{`${translation("Common.title.hello")} ${translation("Common.roles.admin")}`}</PageTitle>
            <Typography variant="headerExtraLargeBold">{translation("Common.title.pleaseSelectMenu")} </Typography>
          </Stack>
          <Grid container padding="0px 40px" spacing="24px">
            {config.home.map((item, index) => (
              <Grid key={index} item width="264px">
                <Card onClick={() => handleClickMenuItem(item)}>
                  <IconAiInterpret />
                  <Typography variant="bodyBold" marginTop="16px">
                    {item.title}
                  </Typography>
                  <Typography variant="bodySmall" color="text.medium" marginTop="4px">
                    {item.description}
                  </Typography>
                </Card>
              </Grid>
            ))}
          </Grid>
        </DashboardPage>
      )}
    </>
  );
};
export default HomePageModule;
