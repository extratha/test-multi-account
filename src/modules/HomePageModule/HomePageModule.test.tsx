import * as RemoteConfig from "firebase/remote-config";

import { mockDashboardMenuConfigResult } from "@/__mocks__/data";
import { spyUseRouter } from "@/testUtils/navigation";
import { flushPromise, render } from "@/testUtils/testUtils";
import HomePageModule from ".";

describe("HomePageModule", () => {
  beforeEach(() => {
    spyUseRouter();

    jest.spyOn(RemoteConfig, "getValue").mockImplementation((_, key): any => {
      if (key === "DashboardMenu") {
        return { asString: () => JSON.stringify(mockDashboardMenuConfigResult) };
      }
    });
  });

  const renderHomePageModule = async () => {
    const view = render(<HomePageModule />);
    await flushPromise();
    return view;
  };

  it("should render correctly", async () => {
    const { asFragment } = await renderHomePageModule();
    expect(asFragment()).toMatchSnapshot();
  });
});
