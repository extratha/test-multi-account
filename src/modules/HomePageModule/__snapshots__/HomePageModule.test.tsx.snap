// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePageModule should render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 24px;
  overflow: hidden;
}

.emotion-1 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  color: unset;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  overflow: auto;
  border-radius: 16px;
  box-shadow: none;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding: 40px;
}

.emotion-3>:not(style):not(style) {
  margin: 0;
}

.emotion-3>:not(style)~:not(style) {
  margin-top: 16px;
}

.emotion-4 {
  margin: 0;
  font-family: fontFamily;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  font-size: 36px;
  font-weight: 700;
  line-height: 1.5;
  background: linear-gradient(94deg, #0059D6 0%, #028DF5 50%, #00CBDD 100%);
  -webkit-background-clip: text;
  background-clip: text;
  -webkit-text-fill-color: transparent;
  text-fill-color: transparent;
  line-height: 1.5;
}

.emotion-5 {
  margin: 0;
  font-size: 36px;
  font-weight: 700;
  line-height: 1.5;
}

.emotion-6 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-top: -24px;
  width: calc(100% + 24px);
  margin-left: -24px;
  padding: 0px 40px;
}

.emotion-6>.MuiGrid-item {
  padding-top: 24px;
}

.emotion-6>.MuiGrid-item {
  padding-left: 24px;
}

.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 264px;
}

.emotion-8 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  color: unset;
  overflow: hidden;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  padding: 16px;
  box-shadow: none;
  border-radius: 12px;
  border: 1px solid #CFD8DC;
}

.emotion-9 {
  margin: 0;
  font-size: 16px;
  font-weight: 700;
  line-height: 1.5;
  margin-top: 16px;
}

.emotion-10 {
  margin: 0;
  font-size: 14px;
  line-height: 1.5;
  color: #607D8B;
  margin-top: 4px;
}

<div
    class="MuiStack-root emotion-0"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 emotion-1"
    >
      <div
        class="MuiStack-root emotion-2"
      >
        <div
          class="MuiStack-root emotion-3"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-4"
          >
            สวัสดี แอดมิน
          </p>
          <h1
            class="MuiTypography-root MuiTypography-headerExtraLargeBold emotion-5"
          >
            โปรดเลือกเครื่องมือที่คุณต้องการ 
          </h1>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container emotion-6"
        >
          <div
            class="MuiGrid-root MuiGrid-item emotion-7"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-8"
            >
              <svg />
              <p
                class="MuiTypography-root MuiTypography-bodyBold emotion-9"
              >
                Title 0
              </p>
              <p
                class="MuiTypography-root MuiTypography-bodySmall emotion-10"
              >
                Description 0
              </p>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item emotion-7"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-8"
            >
              <svg />
              <p
                class="MuiTypography-root MuiTypography-bodyBold emotion-9"
              >
                Title 1
              </p>
              <p
                class="MuiTypography-root MuiTypography-bodySmall emotion-10"
              >
                Description 1
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
